src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -280,5 +282,45 @@ public void SetThreadPrincipal(IPrincipal principal)
                 _defaultPrincipal = principal;
             }
         }
+
+        public IPrincipal GetThreadPrincipal()
+        {
+            IPrincipal principal = null;
+            if (_defaultPrincipal != null)
+            {
+                principal = _defaultPrincipal;
+            }
+            else
+            {
+                switch (_principalPolicy)
+                {
+                    case PrincipalPolicy.NoPrincipal:
+                        principal = null;
+                        break;
+                    case PrincipalPolicy.UnauthenticatedPrincipal:
+                        Assembly assembly = Assembly.Load(new AssemblyName("System.Security.Claims"));
+                        Type genericPrincipal = assembly?.GetType("System.Security.Principal.GenericPrincipal");
```

Comment:
These should not be with`?` mark. If the type does not exist, it should throw.

Reply:
should we throw a certain exception if type is not found ? currently it will just throw nullReferenceException if we remove the ```?```

Reply:
`Assembly.Load` will throw by default, it never returns null.
`GetType` will throw if you set `throwOnError: true`.

Whatever exception these throw is fine.